actionpack/lib/abstract_controller/base.rb

```diff
@@ -16,28 +16,21 @@ def initialize(message = nil, controller = nil, action = nil)
       super(message)
     end
 
-    class Correction
-      def initialize(error)
-        @error = error
-      end
+    if defined?(DidYouMean::Correctable) && defined?(DidYouMean::Jaro)
+      include DidYouMean::Correctable
 
       def corrections
-        if @error.action
-          maybe_these = @error.controller.class.action_methods
+        if action
+          maybe_these = controller.class.action_methods
 
           maybe_these.sort_by { |n|
-            DidYouMean::Jaro.distance(@error.action.to_s, n)
+            DidYouMean::Jaro.distance(action.to_s, n)
           }.reverse.first(4)
         else
           []
         end
       end
```

Comment:
You can use `DidYouMean::SpellChecker` to make the accuracy high enough. The jaro distance isn't accurate and shouldn't be used unless you have a reason to do so. Same applies to other places too.

```ruby
def corrections
  @corrections ||= DidYouMean::SpellChecker.new(dictionary: controller.class.action_methods).correct(action.to_s)
end
```

Reply:
Yes, I want to merge those changes you made in a seperate PR to make this one not too big.

Reply:
@p8 Can you apply the suggestion from @yuki24 in this PR? I'd not think to be worth to spend rounds on this kind of code cleanup.

Reply:
I can see the benefits of both, while this PR already adds value and is changeset+comments are large enough to make additional changes (arguably) harder to review. It does cut down on the noise to include those changes in a single PR.

One alternative(?) suggestion would be to open a PR against this as one way to review the additional changes in isolation.

I don't have a strong opinion either way, just trying to offer some thoughts. Perhaps implementing this suggestion introduces more code to review or more edge-cases to consider -- for example. If I am reviewing many PRs with my limited availability, smaller PRs that add value are easier for me to manage vs. lengthy PRs with a lot of discussion those will take more time to read and investigate.

Reply:
I'll add the changes, no problem 😄

Reply:
@zzak @kamipo I've added Yuki's suggestions.